data/src/test/java/com/microsoft/azure/kusto/data/ConnectionStringBuilderTest.java

```diff
@@ -158,4 +159,13 @@ void createWithAadTokenProviderAuthentication(){
         Assertions.assertDoesNotThrow( () -> ConnectionStringBuilder
                 .createWithAadTokenProviderAuthentication("resource.uri", () -> "token"));
     }
+
+    @Test
+    @DisplayName("validate ClientImpl strips fed=true")
+    void stripFedereatedAuthFromCSB() throws URISyntaxException {
```

Comment:
Since we're trying to get all our code more into testing, I suggest also adding a negative test for this field.

Reply:
what could be a negative test here?

Reply:
void dontChangeUrlIfNotFederated() throws URISyntaxException {
        String clusterUrl = "https://service.uri";
        ConnectionStringBuilder csb = ConnectionStringBuilder.createWithAadApplicationCredentials(clusterUrl, "id", "appKey");
        ClientImpl client = new ClientImpl(csb);
        Assertions.assertEquals(clusterUrl, csb.getClusterUrl());
    }